app/server/lib/shutdown.js

```diff
@@ -81,7 +83,9 @@ function signalExit(signal) {
     log.info("Server %s exiting on %s", prog, signal);
     process.removeListener(signal, dup);
     delete signalsHandled[signal];
-    process.kill(process.pid, signal);
+    // Exit with the expected exit code for being killed by this signal.
```

Comment:
The comment line `70` above probably needs to be updated since the logic has changed.
> It runs the cleanup handlers, and then re-sends the same signal, which will no longer get caught.

Reply:
Thanks @jonathanperret!

I have proposed a change above, does it seem more accurate?

Reply:
This looks good. Would you mind adding an explanation why this approach is used? Like "Unlike re-sending the same signal, the explicit exit works even in a situation when Grist is the init (pid 1) process in a container."

Reply:
Thanks @dsagal, that's done!